bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/BusyIndicator.java

```diff
@@ -70,6 +70,41 @@ public static void showWhile(Display display, Runnable runnable) {
 		}
 	}
 
+	/**
+	 * If called from a {@link Display} thread, waits for the given
+	 * <code>Future</code> to complete and provides busy feedback using the
+	 * busy indicator. While waiting for completion, pending UI events are processed
+	 * to prevent UI freeze.
+	 *
+	 * If there is no {@link Display} for the current thread, the method returns
+	 * immediately and no feedback is shown to the user.
+	 *
+	 * In any case, the passed {@link Future} is returned for chaining of further operations.
+	 *
+	 * @param future the {@link Future} for which busy feedback is to be shown.
+	 * @return the {@link Future} for chaining.
+	 * @since 3.127
+	 */
+	public static <F extends Future<T>, T> F showWhile(F future) {
+		Objects.requireNonNull(future);
+		if (!future.isDone()) {
+			Display display = Display.findDisplay(Thread.currentThread());
+			if (display != null) {
+				Integer busyId = setBusyCursor(display);
+				try {
+					while (!future.isDone() && !display.isDisposed()) {
+						if (!display.readAndDispatch()) {
+							Thread.onSpinWait();
```

Comment:
```suggestion
							Thread.yield();
```

to give the future the needed resources to run.

Reply:
Showing a BusyIndicator is only relevant if it can be seen by the user which requires substantial time. Calling the OS is simply a return as soon as the OS detected that there is no other thread that needs a CPU. In contrast repeating calling readAndDispatch() will not allow the cpu to go into sleep mode, so even if the compiler inserts a power safe instruction the CPU will be busy procesing readAndDispatch(). The nanoseconds that you may save by "yield" in no way justifies burning the CPU time with active readAndDispatch() loop and a potential deadlock.

Reply:
I think Display.sleep and Display.wake as a subscription to the given future are the right approach here to avoid excessive looping on the display thread.

Reply:
BTW you can even test that: put a sleep() in the future and watch the process monitor for burning a complete CPU for the display thread. If you want to to save power consumption you have at least to call `Thread.sleep(0)` since even a yield() burns a CPU du to its fast return.